lldb/source/Plugins/LanguageRuntime/Swift/SwiftTask.cpp

```diff
@@ -0,0 +1,53 @@
+#include "SwiftTask.h"
+#include "SwiftLanguageRuntime.h"
+#include "lldb/Target/Process.h"
+
+using namespace llvm;
+using namespace lldb;
+
+lldb_private::ThreadTask::ThreadTask(tid_t tid, addr_t async_ctx,
+                                     ExecutionContext &exe_ctx)
+    : Thread(exe_ctx.GetProcessRef(), tid, true),
+      m_concrete_reg_ctx_sp(exe_ctx.GetFrameSP()->GetRegisterContext()) {
+  m_async_ctx = async_ctx;
+  auto ptr_size = exe_ctx.GetTargetRef().GetArchitecture().GetAddressByteSize();
+  // A simplified description of AsyncContext. See swift/Task/ABI.h
+  // struct AsyncContext {
+  //   AsyncContext *Parent;                    // offset 0
+  //   TaskContinuationFunction *ResumeParent;  // offset 8
+  // };
+  auto resume_offset = ptr_size; // offsetof(AsyncContext, ResumeParent)
+  auto resume_ptr = async_ctx + resume_offset;
+  Status status;
+  m_pc = exe_ctx.GetProcessRef().ReadPointerFromMemory(resume_ptr, status);
```

Comment:
I'm moderately concerned that this ctor may fail.
Should we make the constructor take the continuation ptr directly and instead create a "make thread function" that can fail?

Reply:
That seems fine to me. Note that if `ReadPointerFromMemory` fails, the resulting address is `LLDB_INVALID_ADDRESS`, which subsequent code should handle.